app/models/sfx_data.rb

```diff
@@ -0,0 +1,84 @@
+##
+# Pull full-text source data from the SFX XML endpoint and render it.
+#
+# Initializes with the SFX URL that would normally render the HTML
+# form/links and append a paramter to it that returns XML instead.
+#
+# This class parses that XML and returns it both as ruby objects and an HTML rendering.
+class SfxData
+  FULL_TEXT_SERVICE_TYPE = 'getFullTxt'.freeze
+  SFX_XML_RESPONSE_TYPE = 'sfx.response_type=multi_obj_xml'.freeze
+
+  def initialize(base_sfx_url)
+    @base_sfx_url = base_sfx_url
+  end
+
+  def targets
+    return [] unless sfx_xml
+    @targets ||= sfx_xml.xpath('//target').map do |t|
+      next unless t.xpath('./service_type').try(:text) == FULL_TEXT_SERVICE_TYPE
+
+      Target.new(t)
+    end.compact
+  end
+
+  private
+
+  attr_reader :base_sfx_url
+
+  def sfx_xml
+    return unless sfx_response.success?
+    @sfx_xml ||= begin
+      Nokogiri::XML.parse(sfx_response.body)
+    rescue
```

Comment:
Yes, there are two cases where `sfx_xml` can return `nil`, but the consumer of this method (the `targets` method) checks that and returns an sensical value (`[]`) in that case.

The other options would be to return an empty XML document, but that didn't quite feel right either.